rudder-web/src/main/scala/com/normation/rudder/web/services/ComplianceData.scala

```diff
@@ -95,6 +95,162 @@ object RulePolicyMode {
         }
     }
   }
+
+  def directiveModeOnNode(nodeMode : Option[PolicyMode], globalMode : GlobalPolicyMode)(directiveMode : Option[PolicyMode]) = {
+   globalMode.overridable match {
+      case PolicyModeOverrides.Unoverridable =>
+        (globalMode.mode.name, s"""Rudder's global agent policy mode is set to <i><b>${globalMode.mode.name}</b></i> and is not overridable on a per-node or per-directive basis. Please check your Settings or contact your Rudder administrator.""")
+      case PolicyModeOverrides.Always =>
+        val (mode,expl) = matchMode(nodeMode,globalMode,directiveMode)
+        (mode.name,expl)
+    }
+  }
+  def matchMode(nodeMode : Option[PolicyMode], globalMode : GlobalPolicyMode, directiveMode : Option[PolicyMode]) : (PolicyMode,String) = {
+    (nodeMode,directiveMode) match {
+      case (None,None) =>
+        (globalMode.mode, "This mode is the globally defined default. You can change it in the global <i><b>settings</b></i>.")
+      case (Some(Enforce),Some(Enforce)) =>
+        (Enforce, "<b>Enforce</b> is forced by both this <i><b>Node</b></i> and this <i><b>Directive</b></i> mode")
+      case (Some(Enforce),None) =>
+        (Enforce, "<b>Enforce</b> is forced by this <i><b>Node</b></i> mode")
+      case (None,Some(Enforce)) =>
+        (Enforce, "<b>Enforce</b> is forced by this <i><b>Directive</b></i> mode")
+      case (Some(Audit), Some(Audit)) =>
+        (Audit, "<b>Audit</b> mode is forced by both this <i><b>Node</b></i> and this <i><b>Directive</b></i>")
+      case (Some(Audit), None) =>
+        (Audit, "<b>Audit</b> mode is forced by this <i><b>Node</b></i>")
+      case (Some(Audit), Some(Enforce)) =>
+        (Audit, "The <i><b>Directive</b></i> is configured to <b class='text-Enforce'>enforce</b> but is overriden to <b>audit</b> by this <i><b>Node</b></i>. ")
+      case (None,Some(Audit))  =>
+        (Audit, "<b>Audit</b> mode is forced by this <i><b>Directive</b></i>")
+      case (Some(Enforce),Some(Audit))  =>
+        (Audit, "The <i><b>Node</b></i> is configured to <b class='text-Enforce'>enforce</b> but is overriden to <b>audit</b> by this <i><b>Directive</b></i>. ")
+    }
+  }
+
+  // The goal of that function is to compute the effective mode applied by a combination of a element of one type and a set of element of another type
+  // and explain why it's result. (ie : Compute the policy mode applied on a RUle depending on mode of One Node and all Directive from that Rule
+
+  // It uses:
+  // * The global defined mode
+  // * A unique policy mode of one element of type A as base (generally a Directive or a Node)
+  // * multiples Policy modes of a set elements of type B that we will look into it to understand what's going on
+  private[this] def generic(
+      uniqueMode : Option[PolicyMode]
+    , uniqueKind : String
```

Comment:
please, can you explain in the comments what are uniqueKind and multipleKind ? this is not straightforward